### ISSUE TITLE
The accepted parameters for LightGBMClassifier

### ISSUE THREAD

**Issue by ywskycn** [Azure/mmlspark#314]
Currently LightGBMClassifier only accepts a subset LightGBM params. Params in https://github.com/Azure/mmlspark/blob/master/src/lightgbm/src/main/scala/LightGBMParams.scala compared to https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst. Just would like to check whether there is a reason behind to put a subset there? I was trying to improve LightGBMClassifier result accuracy recently, and would like to play with more configs. We may extend existing params set.

**Comment by imatiach-msft** [comment]
Hi @ywskycn,
We can extend the existing parameter set if there are more useful parameters for the classifier/regressor.  In fact, @ekaterina-sereda-rf  merged in some additional parameters here:
https://github.com/Azure/mmlspark/pull/282
If you have any additional parameters you would like to add please let us know (or you can send out a similar PR).

**Comment by ywskycn** [comment]
Thanks @imatiach-msft . I was playing around with some internal datasets, but found LightGBM didn't deliver good result. I think I didn't tune some good parameters. BTW, have u tried LightGBMClassifier on some public datasets? If so, it would be great share some parameters and I can try to run in my cluster, just to make sure I config everything correctly.

**Comment by imatiach-msft** [comment]
@ywskycn yes, we have some examples here for classifier and regressor as part of our test sets:
https://github.com/Azure/mmlspark/blob/master/src/lightgbm/src/test/resources/benchmarks_VerifyLightGBMClassifier.csv
https://github.com/Azure/mmlspark/blob/master/src/lightgbm/src/test/resources/benchmarks_VerifyLightGBMRegressor.csv

The AUC is almost always better than that for all spark ML learners (note there is no significant tuning however):
https://github.com/Azure/mmlspark/blob/master/src/train-classifier/src/test/resources/benchmarks_VerifyTrainClassifier.csv


I also did a comparison on Higgs 4GB dataset on Azure Databricks and HDI (note this is not checked in), and it matched the performance of LightGBM:
https://github.com/imatiach-msft/mmlspark/tree/ilmat/higgs-notebooks/notebooks/higgs
You can see metrics here:
https://github.com/guolinke/boosting_tree_benchmarks/blob/master/result.md
You can get the Higgs csv dataset here:
https://github.com/guolinke/boosting_tree_benchmarks/tree/master/data
Specifically this link:
https://archive.ics.uci.edu/ml/datasets/HIGGS

Please also take a look at this related issue, it looks like they had success setting the params below to get the same CPU usage as XGBoost (they didn't mention performance however):
https://github.com/Azure/mmlspark/issues/292

"These are the params I used today:

      .setBaggingFraction(0.8)
      .setFeatureFraction(0.8)
      .setMaxBin(1024)
      .setBaggingFreq(1)
      .setBaggingSeed(1)
      .setMaxDepth(30)
      .setMinSumHessianInLeaf(5000)
      .setLabelCol(labelCol)
      .setApplication("regression_l2")
      .setFeaturesCol(featuresColumn)
      .setNumIterations(num_iterations)
      .setNumLeaves(216)
      .setTimeout(300)
      .setLearningRate(0.01)
      .setPredictionCol(predCol)

This is pretty comparable to what we are getting from xgboost.
"

**Comment by imatiach-msft** [comment]
Can you also send us some of the tests you have done?  I would be interested in taking a look as to why you are not getting good results in some cases.

**Comment by ywskycn** [comment]
@imatiach-msft Thanks for the datasets and detailed explanation, I'll try it out locally.
For the tests I did before, is a simple internal dataset and I run classification using both XGBoost and LightGBM (mostly just use default params). The result AUC is largely different (0.99 v.s 0.75). I guess I need to tune some more parameters there. But I would like to reproduce the job on your provided datasets first, to make sure my local code is correct.

**Comment by imatiach-msft** [comment]
@ywskycn that is concerning, would you be able to share your dataset or a similar dataset that has poor AUC with the corresponding notebook?  I would like to take a look to see if I can somehow improve the results.  It may be the parameters or it could be something about the way you are preprocessing the data.

**Comment by ywskycn** [comment]
Close for now. May add PR for adding additional params in the future if needed.
